prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -287,9 +287,17 @@ ssize_t rxr_pkt_post_ctrl_once(struct rxr_ep *rxr_ep, int entry_type, void *x_en
 	 * if inject, there will not be completion, therefore tx_pkt_entry has to be
 	 * released here
 	 */
-	if (inject)
+	if (inject) {
 		err = rxr_pkt_entry_inject(rxr_ep, pkt_entry, addr);
-	else if (pkt_entry->send->iov_count > 0)
+		/* Currently only EOR packet is injected using shm ep
```

Comment:
Not sure why the shm ep is relevant in this comment or what this means?

Reply:
Actually, this comment is within the function `rxr_pkt_entry_inject()`, and that function is ONLY used by shm EP EOR packet.

Reply:
But this function is general though. What if it's used by something else in the future? Then we'll have a bug here.